test/serialization_test.dart

```diff
@@ -0,0 +1,312 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// serialization_test.dart
+// Tests for serializer and deserializer components
+//
+// 2024 August 27
+// Author: desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+//
+// ignore_for_file: invalid_use_of_protected_member
+
+import 'dart:async';
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+import 'package:test/test.dart';
+
+void main() {
```

Comment:
maybe add a test where the serializer and deserializer are connected to each other and drive two sequential chunks of data?  it's quite possible the components would be often used in pairs, and it would be good that the first time that happens isn't by a user :)

Reply:
I have created a test that links a deserializer->serializer->deserializer.